1.EDA/scripts/9.linear_modeling.py

```diff
@@ -0,0 +1,226 @@
+                + [
+                    other_treatment
+                    for other_treatment in df_patient["treatment"].unique()
+                    if other_treatment != "DMSO"
+                ],
+            )
+            for col in df_patient_trt.columns:
+                if col not in metadata_columns:
+                    # Prepare the formula for the linear model
+                    formula = f"{col} ~ C(treatment)"
+                    # Import statsmodels and run the linear model
+                    model = smf.ols(formula=formula, data=df_patient_trt)
+                    results = model.fit()
+
+                    linear_modeling_results_dict["patient"].append(patient)
+                    linear_modeling_results_dict["treatment"].append(combo[1])
+                    linear_modeling_results_dict["feature"].append(col)
+                    linear_modeling_results_dict["rsquared"].append(results.rsquared)
+                    linear_modeling_results_dict["rsquared_adj"].append(
+                        results.rsquared_adj
+                    )
+                    linear_modeling_results_dict["fvalue"].append(results.fvalue)
+                    linear_modeling_results_dict["pvalue"].append(
+                        results.pvalues[f"C(treatment)[T.{combo[1]}]"]
+                    )
+                    linear_modeling_results_dict["coefficient"].append(
+                        results.params[f"C(treatment)[T.{combo[1]}]"].item()
+                    )
+                    linear_modeling_results_dict["intercept"].append(
+                        results.params["Intercept"].item()
+                    )
+    linear_modeling_results_df = pd.DataFrame(linear_modeling_results_dict)
+    # split the feature column into multiple columns
+    linear_modeling_results_df["feature"].str.split("_", expand=True)
+    linear_modeling_results_df[
+        ["Feature_type", "Compartment", "Channel", "Measurement", "Extra_info"]
+    ] = linear_modeling_results_df["feature"].str.split("_", expand=True)
+
+    # if feature type is area shape then make the measurement the channel and
+    # set the channel to None
+    # this because area size shape features are not channel specific
+    linear_modeling_results_df.loc[
+        linear_modeling_results_df["Feature_type"] == "AreaSizeShape", "Measurement"
+    ] = linear_modeling_results_df["Channel"]
+    linear_modeling_results_df.loc[
+        linear_modeling_results_df["Feature_type"] == "AreaSizeShape", "Channel"
+    ] = None
+    # set compartment to None if is adjacent
+    # this is because adjacent features are not compartment specific
+    linear_modeling_results_df.loc[
+        linear_modeling_results_df["Compartment"] == "adjacent", "Compartment"
+    ] = None
+
+    # run FDR on the p-values
+    pvals = linear_modeling_results_df["pvalue"].values
+    _, pvals_fdr, _, _ = multipletests(pvals, method="fdr_bh")
+    linear_modeling_results_df["pvalue_fdr"] = pvals_fdr
```

Comment:
I think could be a one liner like this. could work c: 
```suggestion
linear_modeling_results_df["pvalue_fdr"] = multipletests(linear_modeling_results_df["pvalue"].values, method="fdr_bh")[1]
```